src/Fixer/FunctionNotation/PhpdocToParamTypeFixer.php

```diff
@@ -0,0 +1,458 @@
+                    continue;
+                }
+
+                if (1 < \count($types)) {
+                    continue;
+                }
+
+                if (0 === \count($types)) {
+                    $paramType = '';
+                }
+
+                if (1 === \count($types)) {
+                    $paramType = array_shift($types);
+                }
+
+                $startIndex = $tokens->getNextTokenOfKind($index, ['(']) + 1;
+                $variableIndex = $this->findCorrectVariable($tokens, $startIndex - 1, $paramTypeAnnotation);
+                if (null === $variableIndex) {
+                    continue;
+                }
+
+                if (!('(' === $tokens[$variableIndex - 1]->getContent()) && $this->hasParamTypeHint($tokens, $variableIndex - 2)) {
+                    continue;
+                }
+
+                $this->fixFunctionDefinition(
+                    $paramType,
+                    $tokens,
+                    $variableIndex,
+                    $hasNull,
+                    $hasArray,
+                    $hasIterable,
+                    $hasVoid,
+                    $hasString,
+                    $hasInt,
+                    $hasFloat,
+                    $hasBool,
+                    $hasCallable,
+                    $hasObject
+                );
+            }
+        }
+    }
+
+    /**
+     * Find all the param annotations in the function's PHPDoc comment.
+     *
+     * @param Tokens $tokens
+     * @param int    $index  The index of the function token
+     *
+     * @return Annotation[]
+     */
+    private function findParamAnnotations(Tokens $tokens, $index)
+    {
+        do {
+            $index = $tokens->getPrevNonWhitespace($index);
+        } while ($tokens[$index]->isGivenKind([
+            T_COMMENT,
+            T_ABSTRACT,
+            T_FINAL,
+            T_PRIVATE,
+            T_PROTECTED,
+            T_PUBLIC,
+            T_STATIC,
+        ]));
+
+        if (!$tokens[$index]->isGivenKind(T_DOC_COMMENT)) {
+            return [];
+        }
+
+        $doc = new DocBlock($tokens[$index]->getContent());
+
+        return $doc->getAnnotationsOfType('param');
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param $index
+     * @param $paramTypeAnnotation
+     *
+     * @return null|int
+     */
+    private function findCorrectVariable(Tokens $tokens, $index, $paramTypeAnnotation)
+    {
+        $nextFunction = $tokens->getNextTokenOfKind($index, [[T_FUNCTION]]);
+        $variableIndex = $tokens->getNextTokenOfKind($index, [[T_VARIABLE]]);
+        if (\is_int($nextFunction) && $variableIndex > $nextFunction) {
+            return null;
+        }
+        if (!isset($tokens[$variableIndex])) {
+            return null;
+        }
+        $variableToken = $tokens[$variableIndex]->getContent();
+        Preg::match('/@param\s*[^\s]+\s*([^\s]+)/', $paramTypeAnnotation->getContent(), $paramVariable);
```

Comment:
This regular expression might have unexpected results for tags that use generics, e.g. in `@param array<int, string> $foo` the variable would be `string>`. I suggest adding several cases with generics in the tests.